source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4460,42 +4464,51 @@ export class Source {
 }
 
 export class Media extends CardElement {
-    poster: string;
-    altText: string;
-    sources: Array<Source> = [];
     private _mediaType: string;
     private _mediaElement: HTMLMediaElement;
+	private _playButton: HTMLAnchorElement;
+	private _pauseButton: HTMLAnchorElement;
+	private _resetButton: HTMLAnchorElement;
+	private _poster: Image;
+
     static readonly acceptedMediaTypes = ["audio", "video"];
 
-    private processSources(): Array<Source> {
-        let typeFilter = (t): boolean => {
-            if (Media.acceptedMediaTypes.indexOf(t)) {
-                this._mediaType = t;
-                typeFilter = (tt) => tt === this._mediaType;
-                return true;
-            }
-            return false;
-        }
-        return this.sources.filter((s): boolean => {
-            if (s.mimeType) {
-                let mimeComponents = s.mimeType.split('/');
-                if (mimeComponents.length == 2) {
-                    return typeFilter(mimeComponents[0]);
-                }
-            }
-            return false;
-        });
+    private processSources(): Array<MediaSource> {
+		let goodSources: Array<MediaSource> = [];
+
+		for (let source of this.sources) {
+			let mimeComponents = source.mimeType.split('/');
+
+			if (mimeComponents.length == 2) {
+				if (!this._mediaType) {
+					let index = Media.acceptedMediaTypes.indexOf(mimeComponents[0]);
+
+					if (index >= 0) {
+						this._mediaType = Media.acceptedMediaTypes[index];
+					}
+				}
+
+				if (mimeComponents[0] == this._mediaType) {
+					goodSources.push(source);
+				}
+			} 
+		}
+
+		return goodSources;
     }
 
-    private makeButton(label: string): HTMLAnchorElement {
+    private static makeButton(label: string, buttonHint: string, handler: ()=>void): HTMLAnchorElement {
         let button = document.createElement("a");
-        button.className = "ac-media-button";
+        button.classList.add("ac-media-button", buttonHint);
```

Comment:
You will need to build the class name by calling HostConfig.makeCssClassName, which means you will need a reference to the button owner's HostConfig. This is necessary so that multiple cards rendered on the same screen can be styled differently using separate stylesheets.